src/Nethermind/Nethermind.JsonRpc.Test/Modules/TestRpcBlockchain.cs

```diff
@@ -244,5 +244,24 @@ await base.Build(builder =>
 
         public Task<string> TestEthRpc(string method, params object?[]? parameters) =>
             RpcTest.TestSerializedRequest(EthRpcModule, method, parameters);
+
+        private IBlockchainProcessor? _currentBlockchainProcessor;
+
+        public async Task RestartBlockchainProcessor()
+        {
+            if (_currentBlockchainProcessor is not null)
+            {
+                await _currentBlockchainProcessor.StopAsync();
+            }
+            else
+            {
+                await BlockchainProcessor.StopAsync();
+            }
+
+            // simulating restarts - we stopped the old blockchain processor and create the new one
+            _currentBlockchainProcessor = new BlockchainProcessor(BlockTree, BlockProcessor,
```

Comment:
shouldn't it support different `IBlockchainProcessor` types? (We really have one major now, so maybe not)

Reply:
Its probably fine for this specific test suite. Otherwise, I'd have to make sure blockchain processor is restartable just for this test.